test/src/test/java/hudson/tasks/ArtifactArchiverTest.java

```diff
@@ -248,6 +265,125 @@ public void testDefaultExcludesOff() throws Exception {
         assertTrue(artifacts.child("dir").child(".svn").child("file").exists());
     }
 
+    @Test
+    @Issue("JENKINS-12379")
+    public void testBasePathOK() throws Exception {
+        FreeStyleProject p = j.createFreeStyleProject();
+
+        p.getBuildersList().replaceBy(Collections.singleton(new CreateFilesForBasePathTest()));
+
+        ArtifactArchiver archiver = new ArtifactArchiver("**");
+        archiver.setBasePath("module/dist");
+        p.getPublishersList().replaceBy(Collections.singleton(archiver));
+
+        assertEquals(Result.SUCCESS, build(p));
```

Comment:
`j.assertBuildStatus`  - and please don't use raw junit assertions without a message